File: Sources/ClaudeRelayClient/Views/AgentDisplayName.swift
import ClaudeRelayKit

/// Presentation-only friendly names for coding agents, layered over the
/// `CodingAgent` registry. The registry stays the data source; this only
/// prettifies for display (e.g. "opencode" → "Open Code").
public enum AgentDisplayName {
    public static func friendly(_ agentId: String?) -> String? {
        guard let agentId else { return nil }
        switch agentId {
        case "opencode": return "Open Code"
        default: return CodingAgent.find(id: agentId)?.displayName ?? agentId
        }
    }
}

File: ClaudeRelayAndroid/feature-workspace/src/main/kotlin/relay/feature/workspace/ui/AgentDisplayName.kt
package relay.feature.workspace.ui

/**
 * Presentation-only friendly names for coding agents. Mirrors the Swift
 * `AgentDisplayName.friendly` (opencode → "Open Code"). Kept in lockstep with
 * the `CodingAgent` registry display names.
 */
fun friendlyAgentName(agentId: String?): String? = when (agentId) {
    null -> null
    "claude" -> "Claude Code"
    "codex" -> "Codex"
    "opencode" -> "Open Code"
    else -> agentId
}
